internal/app/networkd/pkg/networkd/netconf.go

```diff
@@ -55,8 +55,10 @@ func buildOptions(device config.Device, hostname string) (name string, opts []ni
 
 			opts = append(opts, nic.WithNoAddressing())
 		} else {
-			d := &address.DHCP{}
-			opts = append(opts, nic.WithAddressing(d))
+			// No CIDR and DHCP==false results in a static without an IP.
```

Comment:
I know we talked about having another default case here that still defaulted to dhcp, but this code isn't actually called anywhere but here https://github.com/talos-systems/talos/blob/master/internal/app/networkd/pkg/networkd/networkd.go#L77. And so things that are only explicitly defined in the config file will make it here.